homeassistant/components/acmeda/helpers.py

```diff
@@ -0,0 +1,81 @@
+"""Helper functions for Acmeda Pulse."""
+from homeassistant import config_entries
+from homeassistant.helpers.device_registry import async_get_registry as get_dev_reg
+from homeassistant.helpers.entity_registry import async_get_registry as get_ent_reg
+
+from .const import DOMAIN, LOGGER
+
+
+async def update_entities(
+    hass, entity_class, config_entry, current, async_add_entities
+):
+    """Add any new entities, remove any old ones."""
+    if config_entry.entry_id not in hass.data[DOMAIN]:
+        LOGGER.error("Invalid config_entry %s", config_entry.entry_id)
+        return
+
+    hub = hass.data[DOMAIN][config_entry.entry_id]
+    LOGGER.debug("Looking for new %s on: %s", entity_class.__name__, hub.host)
+
+    api = hub.api.rollers
+
+    new_items = []
+    for unique_id, roller in api.items():
+        if unique_id not in current:
+            LOGGER.debug("New %s %s", entity_class.__name__, unique_id)
+            new_item = entity_class(hass, roller)
+            current[unique_id] = new_item
+            new_items.append(new_item)
+
+    async_add_entities(new_items)
+
+    LOGGER.debug("Looking for removed %s on: %s", entity_class.__name__, hub.host)
+
+    removed_items = []
+    for unique_id, element in current.items():
+        if unique_id not in api:
+            LOGGER.debug("Removing %s %s", entity_class.__name__, unique_id)
+            removed_items.append(element)
+
+    for entity in removed_items:
+        # Device is removed from Pulse, so we remove it from Home Assistant
+
+        await entity.async_remove()
```

Comment:
Signal the entity to remove itself via our dispatch helper instead.

Reply:
Done, moved removal of entity and de-registration from device and entity registry to the base class. This is signalled using the dispatch helper as suggested.